### PR TITLE
Update static analyze tools

### PR DESCRIPTION
**Pull request by sidz**
This PR update psalm and phpstan and fixing some founded violations

### CODE REVIEW

File: psalm.xml [hunk 1]
```diff
@@ -1,17 +1,16 @@
 <?xml version="1.0"?>
 <psalm
-    errorLevel="8"
+    errorLevel="7"
     resolveFromConfigFile="true"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xmlns="https://getpsalm.org/schema/config"
     xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
-    errorBaseline="psalm-baseline.xml"
```
Comment (sidz) [inline psalm.xml:8]: all issues has been fixed

File: psalm.xml [hunk 2]
```diff
@@ -1,17 +1,16 @@
 <?xml version="1.0"?>
 <psalm
-    errorLevel="8"
+    errorLevel="7"
```
Comment (sidz) [inline psalm.xml:3]: wondered why we use the lowest possible error level in psalm. So bump it up to 7 and fixed couple new violations

File: psalm.xml [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
     resolveFromConfigFile="true"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xmlns="https://getpsalm.org/schema/config"
     xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
-    errorBaseline="psalm-baseline.xml"
 >
     <projectFiles>
-        <directory name="src/Mutator" />
```
Comment (sidz) [inline psalm.xml:11]: didn't realize why we scan src/Mutator directory only

File: src/Mutator/Arithmetic/Assignment.php [hunk 4]
```diff
@@ -68,8 +68,6 @@ public static function getDefinition(): ?Definition
     }
 
     /**
-     * @psalm-mutation-free
```
Comment (sidz) [inline src/Mutator/Arithmetic/Assignment.php:71]: all `@psalm-mutation-fee` generates
```
ERROR: ImpureMethodCall - src/Mutator/Arithmetic/MinusEqual.php:76:75 - Cannot call an possibly-mutating method PhpParser\Node::getAttributes from a mutation-free context (see https://psalm.dev/203)
        yield new Node\Expr\AssignOp\Plus($node->var, $node->expr, $node->getAttributes());
```

File: src/TestFramework/AbstractTestFrameworkAdapter.php [hunk 5]
```diff
@@ -97,21 +97,21 @@ public function getInitialTestRunCommandLine(
...
      *
-     * @param TestLocation[] $tests
+     * @param TestLocation[] $coverageTests
      *
      * @return string[]
      */
     public function getMutantCommandLine(
-        array $tests,
-        string $mutantFilePath,
+        array $coverageTests,
```
Comment (sidz) [inline src/TestFramework/AbstractTestFrameworkAdapter.php:105]: psalm on level 7 generates an error:
```
ERROR: ParamNameMismatch - src/TestFramework/AbstractTestFrameworkAdapter.php:105:15 - Argument 1 of Infection\TestFramework\AbstractTestFrameworkAdapter::getMutantCommandLine has wrong name $tests, expecting $coverageTests as defined by Infection\AbstractTestFramework\TestFrameworkAdapter::getMutantCommandLine (see https://psalm.dev/230)
        array $tests,
```

that is why property has been renamed. (the next renamed for the same reason)

File: composer.json [hunk 6]
```diff
@@ -67,8 +67,7 @@
         "webmozart/path-util": "^2.3"
     },
     "conflict": {
-        "phpunit/php-code-coverage": ">9 <9.1.4",
-        "symfony/console": "=4.1.5"
```
Comment (sidz) [inline composer.json:71]: no longer required as we supports `symfony/console:^4.1.19`

**Review by maks-rafalko — APPROVED** [review]

File: src/Mutator/Arithmetic/Assignment.php (hunk 4 above)
Comment (sanmai) [reply src/Mutator/Arithmetic/Assignment.php:71]: Let's recall we've added these in #1382 on purpose because we have very annoying and hard to debug bugs.

**Review by sanmai — CHANGES_REQUESTED** [review]
`@psalm-mutation-fee` are there for a reason

**Comment by sidz** [comment]
@sanmai unfortunately https://github.com/infection/infection/pull/1382 doesn't have any explanation about which kind of issues this  annotation fixes. 

p.s. I've put `@psalm-mutation-free` back.

**Comment by maks-rafalko** [comment]
@sidz this comment explains the reasons https://github.com/infection/infection/issues/1380#issuecomment-717804872, this change shows the bug: https://github.com/infection/infection/pull/1382/files#diff-d275481b2f5803179e7bc177f10fd78be286885b52cb0b9922ebe51c125f5137L81


we can add explanation (and probably a link?) to the Mutator interface near `@psalm-mutation-free` to explain why it's needed, here https://github.com/infection/infection/blob/8c97e07391b9f9b7cf977a62616fc458d435edb1/src/Mutator/Mutator.php#L53

**Comment by sidz** [comment]
@maks-rafalko added

File: .php-cs-fixer.dist.php [hunk 7]
```diff
@@ -63,7 +63,6 @@
     ])
     ->ignoreDotFiles(false)
     ->name('*php')
-    ->name('.php_cs.dist')
```
Comment (kubawerlos) [inline .php-cs-fixer.dist.php:66]: This was for PHP CS Fixer 2, can you add `->append([__FILE__])`, so the config is self checked.

**Review by kubawerlos — CHANGES_REQUESTED** [review]

**Comment by sidz** [comment]
It's time to back to this PR :)
So as a lot of stuff were done by @kubawerlos I updated psalm to the latest version on this moment, bumped up errorLevel to 6 and fix couple of founded issues.

so ready for re-review @maks-rafalko, @sanmai

File: psalm.xml [hunk 8]
```diff
@@ -19,7 +19,15 @@
...
             <errorLevel type="suppress">
                 <file name="src/Mutator/Operator/Concat.php"/>
+
+                <directory name="src/Mutator"/>
             </errorLevel>
         </ImpureMethodCall>
+
+        <ImpureFunctionCall>
+            <errorLevel type="suppress">
+                <directory name="src/Mutator"/>
```
Comment (sanmai) [inline psalm.xml:29]:
```suggestion
                <directory name="src/Mutator/Regex"/>
```

File: psalm.xml (hunk 3 above)
Comment (sanmai) [reply psalm.xml:11]: We didn't use Psalm for anything else but to guard against `@psalm-mutation-free`

**Review by sanmai — APPROVED** [review]
Tested this by reintroducing the issue fixed in 7d42b2b3317871e6 (that's when we started to use Psalm). It caught it all right.

**Review by maks-rafalko — APPROVED** [review]

**Comment by sidz** [comment]
thanks @maks-rafalko, @sanmai

**Pull request merged**